src/components/StateMachineProperties.tsx

```diff
@@ -43,74 +43,59 @@ const mapStateToProps = (state: XCSpyState): StateMachinePropertiesProps => {
+            const publicMember = componentProperties[currentComponent].stateMachineProperties[stateMachine][id].jsonMessage;
             return JSON.stringify(publicMember);
         },
-        getStateMachineRef: () => {
-            let id = state.stateMachineProperties.id;
+        getStateMachineRef: (): any => {
+            const id = state.stateMachineProperties.id;
             if (!id)
                 return null;
-            let componentProperties = state.components.componentProperties;
-            let currentComponent = state.components.currentComponent;
-            let stateMachine = state.stateMachineProperties.stateMachine;
+            const componentProperties = state.components.componentProperties;
+            const currentComponent = state.components.currentComponent;
+            const stateMachine = state.stateMachineProperties.stateMachine;
             return componentProperties[currentComponent].stateMachineProperties[stateMachine][id].stateMachineRef;
         },
-        getFirstId: (stateMachine) => {
-            let componentProperties = state.components.componentProperties;
-            let currentComponent = state.components.currentComponent;
+        getFirstId: (stateMachine: string): string => {
+            const componentProperties = state.components.componentProperties;
+            const currentComponent = state.components.currentComponent;
             return Object.keys(componentProperties[currentComponent].stateMachineProperties[stateMachine])[0];
         }
 
     };
 };
 
-const mapDispatchToProps = (dispatch): StateMachinePropertiesCallbackProps => {
+const mapDispatchToProps = (dispatch: Dispatch<XCSpyState>): StateMachinePropertiesCallbackProps => {
     return {
-        clearFinalStates: (component, stateMachine) => {
+        clearFinalStates: (component: string, stateMachine: string): void => {
             dispatch(clearFinalStates(component, stateMachine));
         },
-        setStateMachineId: (id) => {
+        setStateMachineId: (id: string): void => {
             dispatch(setStateMachineId(id));
         },
-        hideStateMachineProperties: () => {
+        hideStateMachineProperties: (): void => {
             dispatch(hideStateMachineProperties());
         },
-        updateGraphic: (currentComponent, stateMachine) => {
-            sessionXCSpy.getPromiseCreateSession()
-                .then((session) => {
-                    session.createSubscriber().getSnapshot(currentComponent, stateMachine, (items) => {
-                        for (let i = 0; i < items.length; i++) {
-                            dispatch(updateGraphic(currentComponent, stateMachine, items[i]));
-                        }
-                        console.log(items);
-                    });
-                });
+        snapshot: (currentComponent: string, stateMachine: string): void => {
+            snapshot(dispatch, currentComponent, stateMachine);
```

Comment:
Best practice, it's better to dispatch an action in your ui code instead of calling a direct core function